packages/authorization-updater/src/index.ts

```diff
@@ -1,33 +1,132 @@
+} from "pagopa-interop-models";
+import {
+  AuthorizationService,
+  authorizationServiceBuilder,
+} from "./authorizationService.js";
+
+const getDescriptorFromEvent = (
+  msg: {
+    data: {
+      descriptorId: string;
+      eservice?: EServiceV2;
+    };
+  },
+  eventType: string
+): {
+  eserviceId: EServiceId;
+  descriptor: Descriptor;
+} => {
+  if (!msg.data.eservice) {
+    throw missingMessageDataError("eservice", eventType);
+  }
+
+  const eservice: EService = fromEServiceV2(msg.data.eservice);
+  const descriptor = eservice.descriptors.find(
+    (d) => d.id === msg.data.descriptorId
+  );
 
-async function processMessage({
-  topic,
-  message,
-  partition,
-}: EachMessagePayload): Promise<void> {
-  try {
-    const messageDecoder = messageDecoderSupplier(topic);
-    const decodedMessage = messageDecoder(message);
-
-    // TODO : update authorization to AuthjorizationManagement service
-
-    logger.info(
-      `Authorization updated after ${JSON.stringify(
-        decodedMessage.type
-      )} event - Partition number: ${partition} - Offset: ${message.offset}`
-    );
-  } catch (e) {
-    logger.error(
-      `Error during message handling. Partition number: ${partition}. Offset: ${message.offset}, ${e}`
-    );
+  if (!descriptor) {
+    throw missingMessageDataError("descriptor", eventType);
   }
+
+  return { eserviceId: eservice.id, descriptor };
+};
+
+async function executeUpdate(
+  eventType: string,
+  messagePayload: EachMessagePayload,
+  update: () => Promise<void>
+): Promise<void> {
+  await update();
+  logger.info(
+    `Authorization updated after ${JSON.stringify(
+      eventType
+    )} event - Partition number: ${messagePayload.partition} - Offset: ${
+      messagePayload.message.offset
+    }`
+  );
+}
+
+function processMessage(authService: AuthorizationService) {
+  return async (messagePayload: EachMessagePayload): Promise<void> => {
+    try {
+      const appContext = getContext();
+      appContext.correlationId = uuidv4();
+
+      const messageDecoder = messageDecoderSupplier(messagePayload.topic);
+      const decodedMsg = messageDecoder(messagePayload.message);
+
+      match(decodedMsg)
+        .with(
+          {
+            event_version: 2,
+            type: "EServiceDescriptorPublished",
+          },
+          {
+            event_version: 2,
+            type: "EServiceDescriptorActivated",
+          },
+          async (msg) => {
+            const data = getDescriptorFromEvent(msg, decodedMsg.type);
+            await executeUpdate(decodedMsg.type, messagePayload, () =>
+              authService.updateEServiceState(
+                "ACTIVE",
+                data.descriptor.id,
+                data.eserviceId,
+                data.descriptor.audience,
+                data.descriptor.voucherLifespan
+              )
+            );
+          }
+        )
+        .with(
+          {
+            event_version: 2,
+            type: "EServiceDescriptorSuspended",
```

Comment:
here we need to handle also `EServiceDescriptorArchived`

Reply:
Ok, it should be a new improvement, current version of archive descriptor method have a different behavior: https://github.com/pagopa/interop-be-catalog-process/blob/399373ce579a56895550f1d8d36d8fbc9a53b199/src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala#L839

Reply:
added 👍🏻